### PR TITLE
(@wdio/globals): fix type propagation

### LINKED ISSUE #11020: [🐛 Bug]: Auto-Completion and Type Prediction Issue for Global Values (e.g., browser, $, expect) in TypeScript on v8 with intelij
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md#reporting-new-issues).

### WebdriverIO Version

latest

### Node.js Version

v16.13.1

### Mode

WDIO Testrunner

### Which capabilities are you using?

```typescript
from auto-generated example
TypeScript version: (e.g., 4.9.5)
```

### What happened?

I am encountering an issue related to auto-completion and type prediction for global values such as browser, $, and expect in TypeScript while using WebdriverIO.

Problem:

When typing global values such as browser, $, and expect in my TypeScript files, the auto-completion suggestions do not appear immediately, even though the correct types are recognized. After typing the value and letting it stand, the TypeScript compiler infers the types correctly, but auto-completion does not show the suggestions in real-time.

### What is your expected behavior?

I expect that when typing global values like browser, $, and expect, the TypeScript editor should provide immediate auto-completion suggestions based on the recognized types. This behavior is crucial for improving coding efficiency and reducing potential errors.

### How to reproduce the bug.

I initiated a new WebdriverIO project using the command npm init wdio . to create a fresh project.

I added the following types setting to my tsconfig.json:

```
"types": [
  "@wdio/globals/types",
  "node",
  "expect-webdriverio",
  "@wdio/mocha-framework"
]
```

Upon writing TypeScript code and using global values like browser, $, and expect, I noticed that auto-completion suggestions were not immediately displayed.

However, TypeScript does recognize the correct types for these global values and provides accurate type checking.

### Relevant log output

```typescript
I confirmed that this issue is not specific to my project by creating a fresh WebdriverIO project using "npm init wdio .", 
and the auto-completion issue was still present.
I have tried different approaches, including adjusting tsconfig.json, triple-slash directives, and TypeScript versions, but the auto-completion issue persists.
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Is there an existing issue for this?

- [X] I have searched the existing issues

### PR DESCRIPTION
**Pull request by christian-bromann**
## Proposed changes

fixes #11020

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [x] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have added proper type definitions for new commands (if appropriate)

## Further comments

n/a

### CODE REVIEW

**Comment by shlomo100** [comment]
In light of the upcoming correction, it will be necessary to update the library documentation to reflect the refined approach for importing global types. This new method leverages the @wdio/globals package, which interfaces through the /build/index.d.ts file.

Furthermore, please direct your attention to the specific section within the package.json file, labeled "exports"

`
"exports": {
  ".": [
    {
      "types": "./build/index.d.ts",
      "import": "./build/index.js",
      "require": "./cjs/index.js"
    },
    "./cjs/index.js"
  ],
  // ... other export entries
}
`

I appreciate your attention to these updates and thank you for your commitment to enhancing the overall quality of the library.

**Comment by erwinheitzman** [comment]
Will test this locally, thanks :)

**Comment by erwinheitzman** [comment]
@shlomo100 I have tested the change locally which seems to work fine, can you share what you meant by your previous message? As I'm not sure what you mean by checking the exports and changing the docs.

**Comment by shlomo100** [comment]
@erwinheitzman @christian-bromann 
Thanks for the fix! I've tested it in the current version, and it works as expected. It's important to highlight that without adding the following line to the tsconfig file:
```ts
"types": [
    "@wdio/globals"
]
```
The automatic completion of the browser global values won't function properly. This adjustment is essential for full functionality. Great job on the fix!

**Comment by erwinheitzman** [comment]
Yes but this is a requirement anyway so no further changes are needed :)

File: packages/wdio-globals/src/index.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import '../types.d.ts'
+/// <reference path="../types.d.ts" />
```
Comment (shlomo100) [inline packages/wdio-globals/src/index.ts:1]: @christian-bromann  I see that in the end this part did not come up in the patch for the latest version, why? This is what makes patching work and make types work

Comment (christian-bromann) [reply packages/wdio-globals/src/index.ts:1]: > This is what makes patching work and make types work

I don't think so. If you create a new WebdriverIO project through `npm init wdio .` you will see that types are propagated correctly if the setup follows our [configuration recommendation](https://webdriver.io/docs/typescript). Please provide a reproducible example if you don't get this.

**Pull request merged**